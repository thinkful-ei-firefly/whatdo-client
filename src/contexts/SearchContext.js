import React from 'react'

const SearchContext = React.createContext({
  city: 'Tucker',
  zipCode: '30084',
  weather: [],
  date: new Date(Date.now()),
  events: [],
  savedEvents: [],
  distance: 10,
  categories: '',
  music: false,
  attractions: false,
  bars: false,
  sports: false,
  performingArts: false,
  festivals: false,
  searchedTerms: {
    zipCode: '30084',
    distance: 10,
    date: new Date(Date.now()),
    music: false,
    attractions: false,
    bars: false,
    sports: false,
    performingArts: false,
    festivals: false,
  },
  apiSearch: () => {},
  getSavedEvents: () => {},
  saveEvent: () => {},
  removeEvent: () => {},
  setSearchTerm: () => {},
  setChecked: () => {},
  setZipCode: () => {},
  pageNum: 1,
  pageCount: 1,
  prevPage: () => {},
  nextPage: () => {},
  loading: false,
  toggleLoading: () => {},
  fetchError: false
})



export default SearchContext