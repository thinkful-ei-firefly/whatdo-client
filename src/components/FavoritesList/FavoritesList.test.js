import React from 'react'
import FavoritesList from './FavoritesList'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import SearchContext from '../../contexts/SearchContext'
// import Search from '../Search/Search'

describe('<FavoritesList />', () => {
  const contextValue = {
    getSavedEvents: () => ['events'],
    savedEvents: ['events']
  }
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <SearchContext.Provider value={contextValue}>
        <FavoritesList />
      </SearchContext.Provider>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <SearchContext.Provider value={contextValue}>
          <FavoritesList />
        </SearchContext.Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
