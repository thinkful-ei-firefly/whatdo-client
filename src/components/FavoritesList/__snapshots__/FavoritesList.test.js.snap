// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> renders as expected 1`] = `
<div
  className="MyEventsPage"
>
  <h2>
    My saved events
  </h2>
  <ul>
    <li>
      <div
        className="EventItem__Container"
      >
        <div
          className="EventItem"
        >
          <span />
          <span />
          <span />
          <span />
          <div
            className="EventItems"
          >
            <h3
              className="EventItem__header"
            />
            <h4
              className="EventItem__time"
            >
              , 
              Invalid Date
            </h4>
            <h5
              className="EventItem__address"
            />
            <div>
              <p />
            </div>
            <div
              className="EventItem__learnMore"
            >
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn More
              </a>
            </div>
            <br />
            <div
              className="EventItem__remove"
            >
              <button
                className="EventItem__removebtn"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
